kernel/runtime/cxx/runtime/cxxsupport.cc

```diff
@@ -268,15 +267,27 @@ int myclone(
     ASSERT(tls != nullptr);
     static int nextThread = 1;
 
+    // The compiler expect a kinda strange alignment coming from clone:
+    // -> rsp % 16 must be 8
+    // You can see this also in musl/src/thread/x86_64/clone.s (rsi is stack)
+    // We will use the same trick for alignment as musl libc
+    auto rsp = (uintptr_t(stack) & uintptr_t(-16))-8;
+
     mythos::PortalLock pl(portal); // future access will fail if the portal is in use already
     mythos::ExecutionContext ec(capAlloc());
     if (ptid && (flags&CLONE_PARENT_SETTID)) *ptid = int(ec.cap());
     // @todo store thread-specific ctid pointer, which should set to 0 by the OS on the thread's exit
     // @todo needs interaction with a process internal scheduler or core manager in order to figure out where to schedule the new thread
-    auto res1 = ec.create(kmem).as(myAS).cs(myCS).sched(mythos::init::SCHEDULERS_START + (nextThread++))
-        .prepareStack(stack).startFunInt(func, arg, ec.cap())
-        .suspended(false).fs(tls)
-        .invokeVia(pl).wait();
+    auto res1 = ec.create(kmem)
+      .as(myAS)
+      .cs(myCS)
+      .sched(mythos::init::SCHEDULERS_START + (nextThread++))
```

Comment:
uff, that slipped through previous reviews. after creating enough threads, the operation will fail because it does not refer to a hardware thread any longer.

would be better to start on the same thread and later use thread affinity or migrate to move the thread to where we want it. This is also better inline with other microkernels, where you always create threads on the caller's core.

Reply:
This is not really in the scope of this pull request and I don't want to touch it right know.

I agree that _something_ should be done here. A better strategy would be doing it round-robin, but we still have no way of knowing how many core there are in usermode. We also have no canonic way of changing the affinity, but this is possible to implement (binding EC to other SC) using `mythos_get_pthread_tid` now.

Implementing a _good_ strategy is afaik the domain of @gypsephi, so maybe talk to him about that before making to much changes here.

Reply:
I agree. Let's put a big warning comment into the code here. This `++` is way too inconspicuous for the scale of problems it is causing.